packages/astro/src/vite-plugin-astro-server/response.ts

```diff
@@ -23,6 +23,18 @@ export async function handle404Response(
 	writeHtmlResponse(res, 404, html);
 }
 
+export async function default404Page(
+	{ pathname }: { pathname: string }
+) {
+	return new Response(notFoundTemplate({
+		statusCode: 404,
+		title: 'Not found',
+		tabTitle: '404: Not Found',
+		pathname,
+	}), { status: 404, headers: { 'Content-Type': 'text/html; charset=utf-8' } });
+}
+default404Page.isAstroComponentFactory = true;
```

Comment:
Let's add a comment that explains what this is doing and why

Reply:
```suggestion
// mark the function as an AstroComponentFactory for the rendering internals
default404Page.isAstroComponentFactory = true;
```